src/static/js/apps/ide.js

```diff
@@ -165,16 +169,10 @@ function AppIDE(pushInterval) {
   this._combineText = function(cursor_pos) {
     var base = this._zoneLastVersion.get();
     var modifs = this._changeMemory.get();
-    modifs.map(function(mod) {
-      base = mod.type == CHANGE_RM_TYPE ?
-        base.cutFrom(mod.pos, mod.count):
-        base.insert(mod.content, mod.pos);
-
-      // The delete will need to be thinked a bit more
-      cursor_pos = mod.pos <= cursor_pos ?
-        cursor_pos + (mod.type == CHANGE_RM_TYPE ? 0 : mod.content) :
-        cursor_pos;
-    });
+    for(var i = 0; i < modifs.length; ++i) {
+      base = modifs[i].applyOnText(base);
+      cursor_pos = modifs[i].applyOnPos(cursor_pos);
```

Comment:
@vphilippon  @maym2104 
I need help on this bug since I did not see it in two hours.

In the bug, modifs variable is an array of dictionnaries (type, count, pos) or (type, content, pos)
They should be object of type ObjectAddChange or ObjectRemoveChange implemented in representation.js